company_dependent_flag/tests/test_module.py

```diff
@@ -0,0 +1,20 @@
+# © 2023 David BEAL @ Akretion
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+import xml.dom.minidom as minidom
+
+from odoo.tests import common
+
+
+class Test(common.SavepointCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.env = cls.env(context=dict(cls.env.context, tracking_disable=True))
+        cls.partner_view = cls.env.ref("base.view_partner_form")
+
+    def test_class_company(self):
+        res = self.env["res.partner"]._fields_view_get(view_id=self.partner_view.id)
+        arch = minidom.parseString(res["arch"])
+        for field in arch.getElementsByTagName("phone"):
+            self.assertIn("building", field.getAttribute("class"))
```

Comment:
**issue:** You are looking for a `<phone>` tag while you should be looking for a `<field name="phone"/>` and make sure to have it inside a `<div>` with also a `<span>` inside that same div

Reply:
Fixed, following your suggestion now it searches for the `<span>` and checks if the icon is there.